base/mqueue/metrics.go

```diff
@@ -0,0 +1,22 @@
+package mqueue
+
+type Counter interface {
```

Comment:
Why do we need to make this abstraction ?

Reply:
This avoids adding new dependencies into modules if they are not directly needed here. We can add prometheus library dependency in the same place where metrics module usage is called. This way mqueue does not depend on Prometheus library.